File: app/android/app/src/main/java/com/magicvector/domain/model/user/UserSessionModel.kt
package com.magicvector.domain.model.user

import com.magicvector.utils.json.GsonBean

data class UserSessionModel(
    val userId: Long,
    val account: String,
    val name: String,
    val avatarUrl: String,
    val accessToken: String,
    val password: String = "",
    val isCurrent: Boolean = false,
    val lastLoginAt: Long = 0L
): GsonBean {
    fun isEmpty(): Boolean {
        return userId <= 0L &&
                account.isEmpty()
    }
}

File: demo/flutter/flutternew/lib/domain/model/user_session_model.dart
class UserSessionModel {
  final int userId;
  final String account;
  final String name;
  final String avatarUrl;
  final String accessToken;
  final String password;
  final bool isCurrent;
  final int lastLoginAt;

  const UserSessionModel({
    required this.userId,
    required this.account,
    required this.name,
    required this.avatarUrl,
    required this.accessToken,
    this.password = '',
    this.isCurrent = false,
    this.lastLoginAt = 0,
  });

  bool get isEmpty => userId <= 0 && account.isEmpty;

  UserSessionModel copyWith({
    int? userId,
    String? account,
    String? name,
    String? avatarUrl,
    String? accessToken,
    String? password,
    bool? isCurrent,
    int? lastLoginAt,
  }) {
    return UserSessionModel(
      userId: userId ?? this.userId,
      account: account ?? this.account,
      name: name ?? this.name,
      avatarUrl: avatarUrl ?? this.avatarUrl,
      accessToken: accessToken ?? this.accessToken,
      password: password ?? this.password,
      isCurrent: isCurrent ?? this.isCurrent,
      lastLoginAt: lastLoginAt ?? this.lastLoginAt,
    );
  }
}
